clients/fides-js/__tests__/lib/cookie.test.ts

```diff
@@ -323,3 +338,289 @@ describe("removeCookiesFromBrowser", () => {
     }
   );
 });
+
+describe("transformTcfPreferencesToCookieKeys", () => {
+  it("can handle empty preferences", () => {
+    const preferences: TcfSavePreferences = { purpose_preferences: [] };
+    const expected: TcfCookieConsent = {
+      purpose_preferences: {},
+      special_purpose_preferences: {},
+      feature_preferences: {},
+      special_feature_preferences: {},
+      vendor_preferences: {},
+      system_preferences: {},
+    };
+    expect(transformTcfPreferencesToCookieKeys(preferences)).toEqual(expected);
+  });
+
+  it("can transform", () => {
+    const preferences: TcfSavePreferences = {
+      purpose_preferences: [
+        { id: 1, preference: UserConsentPreference.OPT_IN },
+      ],
+      special_purpose_preferences: [
+        { id: 23, preference: UserConsentPreference.OPT_OUT },
+      ],
+      feature_preferences: [
+        { id: 1, preference: UserConsentPreference.OPT_IN },
+        { id: 2, preference: UserConsentPreference.OPT_OUT },
+      ],
+      special_feature_preferences: [
+        { id: 1, preference: UserConsentPreference.OPT_IN },
+      ],
+      vendor_preferences: [
+        { id: "1111", preference: UserConsentPreference.OPT_OUT },
+      ],
+      system_preferences: [
+        { id: "ctl_test_system", preference: UserConsentPreference.OPT_IN },
+      ],
+    };
+    const expected: TcfCookieConsent = {
+      purpose_preferences: { 1: true },
+      special_purpose_preferences: { 23: false },
+      feature_preferences: { 1: true, 2: false },
+      special_feature_preferences: { 1: true },
+      vendor_preferences: { 1111: false },
+      system_preferences: { ctl_test_system: true },
+    };
+    expect(transformTcfPreferencesToCookieKeys(preferences)).toEqual(expected);
+  });
+});
+
+describe("updateExperienceFromCookieConsent", () => {
+  const baseCookie = makeFidesCookie();
+
+  // Notice test data
+  const notices = [
+    { notice_key: "one" },
+    { notice_key: "two" },
+    { notice_key: "three" },
+  ] as PrivacyExperience["privacy_notices"];
```

Comment:
I normally try to avoid this type of casting if possible, but I've found it's super handy for tests where a func might only care about a few attributes so we only have to build mock data to cover those attrs